packages/gatsby/src/utils/eslint-config.ts

```diff
@@ -1,26 +1,73 @@
 import { printSchema, GraphQLSchema } from "graphql"
 import { CLIEngine } from "eslint"
 
+const eslintRulePaths = `${__dirname}/eslint-rules`
+const eslintRequirePreset = require.resolve(`./eslint/required`)
+
+export const eslintRequiredConfig: CLIEngine.Options = {
+  rulePaths: [eslintRulePaths],
+  baseConfig: {
+    globals: {
+      graphql: true,
+      __PATH_PREFIX__: true,
+      __BASE_PATH__: true, // this will rarely, if ever, be used by consumers
+    },
+    extends: [eslintRequirePreset],
+  },
+}
+
+export function mergeRequiredConfigIn(
+  existingOptions: CLIEngine.Options
+): void {
+  // make sure rulePaths include our custom rules
+  if (existingOptions.rulePaths) {
+    if (
+      Array.isArray(existingOptions.rulePaths) &&
+      !existingOptions.rulePaths.includes(eslintRulePaths)
+    ) {
+      existingOptions.rulePaths.push(eslintRulePaths)
+    }
+  } else {
+    existingOptions.rulePaths = [eslintRulePaths]
+  }
+
+  // make sure we extend required preset
+  if (!existingOptions.baseConfig) {
+    existingOptions.baseConfig = {}
+  }
+
+  if (existingOptions.baseConfig.extends) {
+    if (
+      Array.isArray(existingOptions.baseConfig.extends) &&
```

Comment:
So if `.extends` exists but is not an array then the `eslintRequirePreset` part is silently dropped? Smells like a bit like a footgun.

Reply:
You are right, I thought `extends` only accept arrays but it also accept just strings, so I will add handling for it

Reply:
Added handling for single string case in https://github.com/gatsbyjs/gatsby/pull/28911/commits/f5ad45a070346ccc8e380180c31b6b71a9ef02ac + some tests for it